UnitystationLauncher/Services/CodeScanService.cs

```diff
@@ -0,0 +1,357 @@
+                CopyFilesRecursively(oldPath,  dataPath.ToString());
+                Directory.Delete(oldPath,true);
+            }
+
+
+            switch (_environmentService.GetCurrentEnvironment())
+            {
+                case CurrentEnvironment.WindowsStandalone:
+                    var exeRename = processingDirectory.GetFiles()
+                        .FirstOrDefault(x => x.Extension == ".exe" && x.Name != "UnityCrashHandler64.exe"); //TODO OS
+
+                    if (exeRename == null || exeRename.Directory == null)
+                    {
+                        errors.Invoke("no Executable found ");
+                        DeleteContentsOfDirectory(processingDirectory);
+                        return false;
+                    }
+                    exeRename.MoveTo(Path.Combine(exeRename.Directory.ToString(), dataPath.Name.Replace("_Data", "") + ".exe"));
+                    break;
+                case CurrentEnvironment.LinuxStandalone:
+                    var ExecutableRename = processingDirectory.GetFiles()
+                        .FirstOrDefault(x => x.Extension == ".x86_64"); 
+
+                    if (ExecutableRename == null || ExecutableRename.Directory == null)
+                    {
+                        errors.Invoke("no Executable found ");
+                        DeleteContentsOfDirectory(processingDirectory);
+                        return false;
+                    }
+                    ExecutableRename.MoveTo(Path.Combine(ExecutableRename.Directory.ToString(), dataPath.Name.Replace("_Data", "") + ".x86_64"));
+                    break;
+            }
+            
+
+            var targetDirectoryinfo = new DirectoryInfo(targetDirectory);
+            if (targetDirectoryinfo.Exists)
+            {
+                DeleteContentsOfDirectory(targetDirectoryinfo);
+            }
+            
+            CopyFilesRecursively(processingDirectory.ToString(), targetDirectory.ToString());
+            DeleteContentsOfDirectory(processingDirectory);
+            DeleteContentsOfDirectory(stagingDirectory); 
+        }
+        catch (Exception e)
+        {
+            errors.Invoke(" an Error happened > " + e);
+            DeleteContentsOfDirectory(processingDirectory);
+            DeleteContentsOfDirectory(stagingDirectory);
+            return false;
+        }
+
+        return true;
+    }
+
+    public string GetManagedOnOS(string GoodFiles)
+    {
+        var OS = _environmentService.GetCurrentEnvironment();
+        switch (OS)
+        {
+            case CurrentEnvironment.WindowsStandalone:
+                return Path.Combine(GoodFiles, "Unitystation_Data", Managed);
+            case CurrentEnvironment.LinuxStandalone:
+                return Path.Combine(GoodFiles, "Unitystation_Data", Managed);
+            case CurrentEnvironment.MacOsStandalone:
+                return Path.Combine(GoodFiles, @"Contents\Resources\Data", Managed);
+        }
+
+        return "idk";
```

Comment:
`CurrentEnvironment.LinuxFlatpak` will return 'idk' here, I think it should be safe to use the same as the `CurrentEnvironment.LinuxStandalone`.